packages/inline-edit/src/InlineEditUncontrolled.tsx

```diff
@@ -0,0 +1,208 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import React, {
+  FC, ReactNode, useRef, useCallback, createRef, useState, useEffect,
+} from 'react';
+
+import {
+  ReadViewWrapper,
+  EditButton,
+  ReadViewContentWrapper,
+  ButtonsWrapper,
+  ButtonWrapper,
+  Button,
+  ContentWrapper,
+} from './styled/InlineEdit';
+import {
+  ConfirmIcon,
+  CancelIcon,
+} from './Icons';
+import { IInlineEditUncontrolledProps } from './interfaces';
+
+
+export const getInlineEditUncontrolled: <TFieldValue = string>() => FC<
+IInlineEditUncontrolledProps<TFieldValue>> = function f<TFieldValue>() {
+  return ({
+    appearance,
+    baseAppearance,
+    defaultValue,
+    readView,
+    editView,
+    readViewFitContainerWidth,
+    isEditing,
+    onEditRequested,
+    onCancel,
+    onConfirm,
+  }: IInlineEditUncontrolledProps<TFieldValue>) => {
+    const editButtonRef = createRef<HTMLButtonElement>();
+    const confirmButtonRef = createRef<HTMLButtonElement>();
+    const cancelButtonRef = createRef<HTMLButtonElement>();
+
+    const startX = useRef(0);
+    const startY = useRef(0);
+    const prevIsEditing = useRef(isEditing);
+    const removeMouseDown: React.MutableRefObject<void | null> = useRef(null);
+
+    const [value, setValue] = useState(defaultValue);
+    const [preventFocusOnEditButton, setPreventFocusOnEditButton] = useState(false);
+
+    useEffect(() => {
+      if (prevIsEditing.current && !isEditing) {
+        if (preventFocusOnEditButton) {
+          setPreventFocusOnEditButton(false);
+        } else if (editButtonRef && editButtonRef.current) {
+          editButtonRef.current.focus();
+        }
+      }
+    }, [editButtonRef, isEditing, preventFocusOnEditButton]);
+
+    useEffect(() => {
```

Comment:
Logic inside this `useEffect `and `removeMouseDown` are looking unnecessary.

Reply:
It is necessary because we have outsideClick event in this case: what we should do if we click not only 'confirm' and 'cancel' button? We should confirm value in edit view component or not? Or we shouldn't do anymore? These questions will be decided in next versions of component, so this logic is necessary for future behavior